src/utils/telemetry.ts

```diff
@@ -207,3 +225,62 @@ export const consoleTelemetryExporter: TelemetryExporter = {
     console.debug('Telemetry Count', count);
   },
 };
+
+/**
+ * A {@link TelemetryExporter} which does nothing with telemetry events
+ */
+export const nullTelemetryExporter: TelemetryExporter = {
+  /**
+   * Does nothing with the timespan event
+   */
+  sendSpan: () => {},
+
+  /**
+   * Does nothing with the count event
+   */
+  sendCount: () => {},
+};
+
+/**
+ * Telemetry entry point to initialize the singleton {@link TelemetryProvider} or
+ * get the current {@link TelemetryProvider}
+ */
+export class Telemetry {
+  private static instance: TelemetryProvider | undefined;
+
+  /**
+   * Sets the singleton {@link TelemetryProvider} to the given provider
+   *
+   * @param provider The {@link TelemetryProvider} for collecting and sending telemetry
+   */
+  static setInstance(provider: TelemetryProvider) {
+    Telemetry.instance = provider;
+  }
+
+  /**
+   * Returns a {@link DefaultTelemetryProvider} which sends telemetry to the given
+   * telemetry exporter
+   *
+   * @param exporter The exporter for sending telemetry
+   */
+  static withExporter(exporter: 'console' | TelemetryExporter | undefined) : TelemetryProvider {
```

Comment:
For a future refactor you might consider defining and `WithExporterParam` type or something like that. Not necessary to update this PR, just remarking.